### PR TITLE
Add new tool - epoch converter

### PR DESCRIPTION
**Pull request by BryanL-Ensemble**
Convert epoch to human-readable date and vice versa

Fix [ENSTI-612](https://jira.ensemble.com/jira/browse/ENSTI-612)

### CODE REVIEW

**Comment by BryanL-Ensemble** [comment]
Demo

epoch-tool-demo

File: src/tools/epoch-converter/epoch-converter.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Convert Epoch to Human Readable Date
+export function epochToDate(epoch: string | number): string {
+  const num = typeof epoch === 'string' ? Number.parseInt(epoch, 10) : epoch;
+
+  if (Number.isNaN(num)) {
+    throw new TypeError('Invalid epoch timestamp');
+  }
+
+  const timestamp = num < 1e12 ? num * 1000 : num;
```
Comment (davidhazama) [inline src/tools/epoch-converter/epoch-converter.service.ts:9]: If we are going to be doing some conversions much like the reference tool, we should include messaging indicating as such and establish proper boundaries of values to evaluate. It would also be good to comment in code which portions map to which measurements of time (ie seconds, milliseconds, microseconds etc)

File: src/tools/epoch-converter/epoch-converter.service.test.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { describe, expect, it } from 'vitest';
+import { dateToEpoch, epochToDate } from './epoch-converter.service';
+
```
Comment (davidhazama) [inline src/tools/epoch-converter/epoch-converter.service.test.ts:3]: It may be good to include boundary value testing for overflows/underflows of possible inputs (ie see the brief excerpt on the reference tool about the potentially troublesome date of January 19, 2038)

File: src/tools/epoch-converter/epoch-converter.vue [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    error.value = e.message;
+  }
+}
+
+function pad(value: string): string {
+  return value.toString().padStart(2, '0');
+}
+</script>
+
+<template>
```
Comment (davidhazama) [inline src/tools/epoch-converter/epoch-converter.vue:47]: I think it would also be good to provide the GMT readable date-time as well as the local one if possible

**Review by davidhazama — CHANGES_REQUESTED** [review]

File: src/tools/epoch-converter/epoch-converter.vue (hunk 3 above)
Comment (BryanL-Ensemble) [reply src/tools/epoch-converter/epoch-converter.vue:47]: Tool now supports both local and GMT readable time

File: src/tools/epoch-converter/epoch-converter.service.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * @param options - Optional flag to interpret the date string as UTC time.
+ * @returns Epoch time as a number in seconds.
+ */
+export function dateToEpoch(
+  dateString: string,
+  options?: {
+    utc?: boolean // if true, force input to be parsed as UTC
+  },
+): number {
+  const normalized = options?.utc && !dateString.endsWith('Z')
```
Comment (davidhazama) [inline src/tools/epoch-converter/epoch-converter.service.ts:59]: In the majority of cases, ternary operators are best used when the evaluations are brief and easy to digest

The use of the word `normalized` is good, but let's break out the considerations - you can still use the short-circuit evaluation but name it something like `shouldNormalizeToUTC` or however you intend to use the `utc` option in the params.

The `utc` param itself may be a bit vague too, as it is unclear if that is the intent to communicate that `dateString` is already in UTC format as the source or if UTC should be the target conversion

**Review by davidhazama — CHANGES_REQUESTED** [review]

File: src/tools/epoch-converter/epoch-converter.vue [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+    epochOutput.value = dateToEpoch(isoString, { parseAsUTC: true }).toString();
+    dateInputError.value = null;
+  }
+  catch (e: any) {
+    epochOutput.value = '';
+    dateInputError.value = e.message;
+  }
+}
+
+function validateDateParts(parts: DateParts): void {
```
Comment (davidhazama) [inline src/tools/epoch-converter/epoch-converter.vue:69]: Is there a way we could leverage a dependency to handle this validation for the parts for us? whether it be part of Javascript native functionality or `moment` or other - reading through, I think it would fail the test for inputting a date of February 31
